### PR TITLE
Updated problem_04 in project_euler

### PR DESCRIPTION
**Pull request by shellhub**
### **Describe your change:**

* [ ] Add an algorithm?
* [x] Fix a bug or typo in an existing algorithm?
* [x] Documentation change?

### **Checklist:**
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Python/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new Python files are placed inside an existing directory.
* [x] All filenames are in all lowercase characters with no spaces or dashes.
* [x] All functions and variable names follow Python naming conventions.
* [x] All function parameters and return values are annotated with Python [type hints](https://docs.python.org/3/library/typing.html).
* [x] All functions have [doctests](https://docs.python.org/3/library/doctest.html) that pass the automated testing.
* [ ] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: project_euler/problem_04/sol1.py [hunk 1]
```diff
@@ -12,6 +12,10 @@ def solution(n):
     """Returns the largest palindrome made from the product of two 3-digit
     numbers which is less than n.
 
+    >>> solution(10000)
+    Traceback (most recent call last):
+    ...
+    ValueError: No such latest palindrome number.
```
Comment (cclauss) [inline project_euler/problem_04/sol1.py:18]: Nice!

Let's move this test below the others so we show how the function works before we show how it breaks,

File: project_euler/problem_04/sol1.py [hunk 2]
```diff
@@ -20,23 +24,28 @@ def solution(n):
...
             divisor = 999
 
             # if 'number' is a product of two 3-digit numbers
             # then number is the answer otherwise fetch next number.
             while divisor != 99:
-                if (number % divisor == 0) and (len(str(int(number / divisor))) == 3):
+                if (number % divisor == 0) and (len(str(number // divisor)) == 3):
                     return number
                 divisor -= 1
+    raise ValueError("No such latest palindrome number.")
```
Comment (cclauss) [inline project_euler/problem_04/sol1.py:43]: This error message is incorrect.  It is not the the number does not exist, it is that it is larger than our acceptable range.

Comment (shellhub) [reply project_euler/problem_04/sol1.py:43]: Can you give an error message ？ Thanks

**Comment by shellhub** [comment]
Why we need change 3 to 3.0 like this 
if (number % divisor == 0) and (len(str(number // divisor)) == 3.0): Thanks 😊

**Review by cclauss — APPROVED** [review]

**Comment by cclauss** [comment]
Given that 3 == 3.0, it is not necessary but...
```
>>> 9 // 3
3
>>> 9.3 // 3.1
3.0
>>> 9 // 3.0
3.0
>>> 9.0 // 3
3.0
```

**Pull request merged**